src/utils/internal.ts

```diff
@@ -813,3 +818,34 @@ export function conditionsAreEqual(
 
   return equalClaims && equalClaimIssuers;
 }
+
+/**
+ * @hidden
+ *
+ * Transforms CaCheckpointTypeParams values to instance either Checkpoint or CheckpointSchedule
+ */
+export async function getCheckpointValue(
+  checkpoint: CaCheckpointTypeParams,
+  token: string | SecurityToken,
+  context: Context
+): Promise<Checkpoint | CheckpointSchedule | Date> {
+  if (
+    checkpoint instanceof Checkpoint ||
+    checkpoint instanceof CheckpointSchedule ||
+    checkpoint instanceof Date
+  ) {
+    assertCaCheckpointValid(checkpoint);
```

Comment:
I am a little unsure about this assertion in a generic conversion function like this. It's asserting a date must be in the future, but I am not sure if that's going to be the case for everything that might call this method.

It might be reasonable that all references to checkpoints by dates will be in the future, so it might be OK to leave in. I think the exists check makes sense in that.

Reply:
The idea was to remove the assertion, since the validation is already being done in the CA creation procedure. The `await` comment was just academic